modules/navier_stokes/src/utils/FaceCenteredMapFunctor.C

```diff
@@ -0,0 +1,132 @@
+//* This file is part of the MOOSE framework
+//* https://www.mooseframework.org
+//*
+//* All rights reserved, see COPYRIGHT for full restrictions
+//* https://github.com/idaholab/moose/blob/master/COPYRIGHT
+//*
+//* Licensed under LGPL 2.1, please see LICENSE for details
+//* https://www.gnu.org/licenses/lgpl-2.1.html
+
+#include "FaceCenteredMapFunctor.h"
+
+template <typename T, typename Map>
+typename FaceCenteredMapFunctor<T, Map>::ValueType
+FaceCenteredMapFunctor<T, Map>::evaluate(const ElemArg & elem_arg, unsigned int) const
+{
+  // The following reconstruction is based on Weller's method. For more information on this,
+  // we recommend:
+  //
+  // Weller, Hilary. "Non-orthogonal version of the arbitrary polygonal C-grid and a new diamond
+  // grid." Geoscientific Model Development 7.3 (2014): 779-797.
+  //
+  // and
+  //
+  // Aguerre, Horacio J., et al. "An oscillation-free flow solver based on flux reconstruction."
+  // Journal of Computational Physics 365 (2018): 135-148.
+  //
+  // This basically reconstructes the cell value based on flux values as follows:
+  //
+  // $\left( \sum_f n_f \outer S_f \right)^{-1} \sum_f (\phi_f \cdot S_f)n_f$
+  //
+  // where $S_f$ is the surface normal vector, $n_f$ is the unit surface vector and $\phi_f$ is a
+  // vector value on the field. Hence the restriction to vector values.
```

Comment:
Again very much appreciate the comment...you're doing a good job of answering my questions as I come up with them!